File: bonsai_interfaces/src/main/java/de/unibi/citec/clf/bonsai/actuators/JointControllerActuator.java

package de.unibi.citec.clf.bonsai.actuators;



import de.unibi.citec.clf.bonsai.core.object.Actuator;
import java.io.IOException;
import java.util.concurrent.Future;

/**
 * Interface to ros control
 * 
 * @author llach
 */
public interface JointControllerActuator extends Actuator{
    
    /**
     * Moves the Z-Lift to given position
     * 
     * @param dist
     * @return
     * @throws IOException 
     */
    Future<Boolean> goToZliftHeight(float dist) throws IOException;

    Future<Boolean> goToHeadPose(float j0, float j1) throws IOException;
    
}

File: bonsai_interfaces/src/main/java/de/unibi/citec/clf/bonsai/actuators/JointControllerActuator.kt
package de.unibi.citec.clf.bonsai.actuators


import de.unibi.citec.clf.bonsai.core.`object`.Actuator
import de.unibi.citec.clf.btl.data.common.Timestamp
import de.unibi.citec.clf.btl.units.TimeUnit
import java.io.IOException
import java.util.concurrent.Future

/**
 * Interface to simple joints
 *
 * @author llach
 */
interface JointControllerActuator : Actuator {

    /**
     * @param duration duration of the move in 1/duration sec
     */

    @Throws(IOException::class)
    fun moveTo(pose: Float, duration: Long, unit: TimeUnit = TimeUnit.MILLISECONDS): Future<Boolean>
    fun moveTo(pose: Float, speed: Double): Future<Boolean>

    fun getMax(): Double?
    fun getMin(): Double?
    fun getPosition(): Double?

}
